client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -15,18 +19,22 @@ private final class PoolManager[F[_], A <: Connection[F]](
     maxTotal: Int,
     maxWaitQueueLimit: Int,
     maxConnectionsPerRequestKey: RequestKey => Int,
+    waitExpiryTime: RequestKey => Int,
     implicit private val executionContext: ExecutionContext)(implicit F: Effect[F])
     extends ConnectionManager[F, A] {
 
-  private sealed case class Waiting(key: RequestKey, callback: Callback[NextConnection])
+  private sealed case class Waiting(
+      key: RequestKey,
+      callback: Callback[NextConnection],
+      at: Instant)
 
   private[this] val logger = getLogger
 
   private var isClosed = false
   private var curTotal = 0
   private val allocated = mutable.Map.empty[RequestKey, Int]
   private val idleQueues = mutable.Map.empty[RequestKey, mutable.Queue[A]]
-  private val waitQueue = mutable.Queue.empty[Waiting]
+  private var waitQueue = mutable.Queue.empty[Waiting]
```

Comment:
Why did this need to become a var?

Reply:
It was a val before and we were dequeuing for all expired one by one. Now we want to use span while taking out expired requests and just assign the new queue. Let me know if you feel this needs to change.